services/thermal_mgr/thermal_mgr.c

```diff
@@ -38,23 +38,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
   thermalMgrQueueHandle = xQueueCreateStatic(
     THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
     thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (event != NULL) {
+    if (xQueueSend(thermalMgrQueueHandle, event, 0 == pdPASS)) {
+      return ERR_CODE_SUCCESS;
+    }
+  }
+
+  return ERR_CODE_UNKNOWN;
 
-  return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t newEvent;
+  newEvent.type = THERMAL_MGR_INTERRUPT_TEST_TEMP_CMD;
+  thermalMgrSendEvent(&newEvent);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_t eventBuffer;
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &eventBuffer, 10) == pdPASS) {
+      float temp;
+      if (readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp) == ERR_CODE_SUCCESS) {
```

Comment:
check the received command before reading the temperature since if the command is something other than measure temp or interrupt like if it's an invalid cmd for example it doesn't make sure to read the temperature anyways